src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -1672,6 +1670,10 @@ void JavaThread::prepare(jobject jni_thread, ThreadPriority prio) {
   // added to the Threads list for if a GC happens, then the java_thread oop
   // will not be visited by GC.
   Threads::add(this);
+  // Publish the JavaThread* in java.lang.Thread after the JavaThread* is
+  // on a ThreadsList. There will be a release when the Theads_lock is
+  // dropped somewhere in the caller, but let's be more proactive.
```

Comment:
Okay. I've updated the comment to:
```
  // Publish the JavaThread* in java.lang.Thread after the JavaThread* is
  // on a ThreadsList. We don't want to wait for the release when the
  // Theads_lock is dropped somewhere in the caller since the JavaThread*
  // is already visible to JVM/TI via the ThreadsList.
```